src/pages/valvonta-kaytto/valvonta-api.js

```diff
@@ -232,22 +232,53 @@ const setFieldToNoneIfNoDocument = field => data => {
   )(data);
 };
 
+const setFieldToNoneIfNoAnswer = field => data => {
+  return R.when(
+    R.both(R.propEq('recipient-answered', false), R.has(field)),
+    R.set(R.lensProp(field), Maybe.None())
+  )(data);
+};
 /**
  * Unwraps the maybe to the value if the field exists
  */
 const unwrapMaybeIfExists = field => data => {
   return R.when(
-    R.has(field),
+    R.both(R.has(field), R.compose(Maybe.isMaybe, R.view(R.lensProp(field)))),
     R.over(R.lensProp(field), Maybe.orSome(null))
   )(data);
 };
 
+/**
+ * Removes all keys from object that have null or undefined values
+ * @param {Object} object
+ */
+const removeNullValues = R.compose(
+  R.fromPairs,
+  R.reject(R.compose(R.isNil, R.prop(1))),
+  R.toPairs
+);
+
 /**
  * Unwraps maybes if document is true for the osapuoli
  */
 export const serializeOsapuoliSpecificData = osapuoliSpecificData => {
   return R.map(
     R.compose(
+      removeNullValues,
+      unwrapMaybeIfExists('recipient-answered'),
+      setFieldToNoneIfNoDocument('recipient-answered'),
+      unwrapMaybeIfExists('answer-commentary-fi'),
+      unwrapMaybeIfExists('answer-commentary-sv'),
+      unwrapMaybeIfExists('statement-fi'),
+      unwrapMaybeIfExists('statement-sv'),
+      setFieldToNoneIfNoAnswer('answer-commentary-fi'),
+      setFieldToNoneIfNoAnswer('answer-commentary-sv'),
+      setFieldToNoneIfNoAnswer('statement-fi'),
+      setFieldToNoneIfNoAnswer('statement-sv'),
+      setFieldToNoneIfNoDocument('answer-commentary-fi'),
+      setFieldToNoneIfNoDocument('answer-commentary-sv'),
+      setFieldToNoneIfNoDocument('statement-fi'),
+      setFieldToNoneIfNoDocument('statement-sv'),
```

Comment:
Näitä voisi ehkä kasata composella yhden kentän funktioiksi ja loopata sitten kaikille kentille mille se operaatio tehdään?

Reply:
Makustelin tätä ja minusta tuntui paremmalta tehdä niin päin, että tein näille funktioille wrapperit jotka ottavat parametrina listan fieldejä ja sitten ajavat niille kaikille nämä alkuperäiset funktiot. Tehtävät operaatiot pysyivät selkeinä ja sitten helpompi määritellä mille kentille tehdään mikäkin. Miltä näyttää sun mielestä tämä toteutus?

Reply:
Selkeämpi ehdottomasti kuin alkuperäinen. Mennään tällä